test/ModuleInterface/no-implicit-extra-clang-opts.swift

```diff
@@ -0,0 +1,25 @@
+// RUN: %empty-directory(%t)
+// RUN: %empty-directory(%t/ModuleCache)
+
+import SIMod
+
+// Step 0: Copy relevant files into the temp dir which will serve as the search path
+// RUN: cp %S/Inputs/implicit-options-inheritance/module.modulemap %t/module.modulemap
+// RUN: cp %S/Inputs/implicit-options-inheritance/CIMod.h %t/CIMod.h
+// RUN: cp %S/Inputs/implicit-options-inheritance/SIMod.swiftinterface %t/SIMod.swiftinterface
+
+// Step 1: Build this file, causing an implicit build of SIMod and CIMod into the module cache.
+// Pass in a clang arg pointing it to a modulemap that has nothing to do with downstream modules and is not on the search path.
+
+// RUN: %target-swift-frontend -emit-module -module-name no-implicit-extra-clang-maps -o %t/no-implicit-extra-clang-maps.swiftmodule %s -I %t -Xcc -fmodule-map-file=%S/Inputs/implicit-options-inheritance/test-dummy.modulemap -module-cache-path %t/ModuleCache
+
+// Step 2: Touch the dummy modulemap we passed in with `-Xcc -fmodule-map-file` above.
+// RUN: touch %S/Inputs/implicit-options-inheritance/test-dummy.modulemap
```

Comment:
This is wrong. One is not allowed to mutate the inputs in any way, including changing modification timestamps. This breaks build reproducibility.

Reply:
It also breaks the ability to have concurrent builds from the same source directory (for example: with and without asserts, with and without optimizations, etc) because now one build can affect the results of the other.